unic/ucd/name/src/name.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2017 The UNIC Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+use std::fmt;
+
+use unic_utils::CharDataTable;
+
+
+#[derive(Copy, Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub struct Name {
+    pieces: &'static [&'static str],
+}
+
+impl Name {
+    pub fn of(ch: char) -> Option<Name> {
+        data::NAMES.find(ch).map(|&pieces| Name { pieces })
```

Comment:
`Option::map` is what we would do anyway:

<https://github.com/rust-lang/rust/blob/d21ec9b4efd1da012979b050bc0a0426fe45fcdf/src/libcore/option.rs#L378-L401>

```rust
    pub fn map<U, F: FnOnce(T) -> U>(self, f: F) -> Option<U> {
        match self {
            Some(x) => Some(f(x)),
            None => None,
        }
    }
```

I agree we'd want to test it, but for the structures that are just simple wrappers, it should be a noop anyway.